### ISSUE TITLE
Allow Request Transformer to dynamically modify upstream API url

### ISSUE THREAD

**Issue by rdzak** [Mashape/kong#2801]
Plugin allows modification of body, query parameters, and headers. It does not allow to modify URL path itself. To use upstream API with URL structure shaped RESTfully (e.g /customer/123/videos/234) where 123 comes from Kong itself (other plugins) is impossible (not original request).

We can dynamically attach 123 as part of query parameters like ?blah=123 for instance but this is not enough when working with modern resource-based RESTful APIs.

Please up-vote this request!

**Comment by p0pr0ck5** [comment]
Hey @rdzak,

Thanks for the submission. The Request Transformer plugin available in [Mashape Enterprise](https://www.mashape.com/enterprise/) does indeed have the functionality! This is also noted in the open source request transformer plugin docs (https://getkong.org/plugins/request-transformer/).

BTW, in the future, please try to reserve GitHub issues for actual issues running Kong, not feature or support requests. This is noted in the boilerplate when you create a new issue. :) We can absolutely continue the ongoing discussion in the mailing list: https://groups.google.com/forum/#!topic/konglayer/b3QAaUdOCb4. Please also try to avoid cross-posting like this, as it tends to cause confusion. Thanks!

**Comment by rdzak** [comment]
Apologies - I have see similar posts here so I followed the pattern. From the note you mentioned in https://getkong.org/plugins/request-transformer/ I do not get that the functionality I described indeed exists. There is no mention of ability to place arbitrary dynamic data (not coming from request content, but from other plugins) into upstream variables like `customers/{123}/videos/345` for `123` being in my case consumer_id (from auth plugin).
